src/api/cart/presentation/cart.controller.rest.ts

```diff
@@ -0,0 +1,142 @@
+import { Request, Response } from 'express';
+import Joi from 'joi';
+
+import { BaseController } from '../../../common';
+import { CartService, cartService } from '../service/cart.service';
+import { HttpStatusCode } from '../../../shared';
+import { AuthMiddleware, ValidateMiddleware } from '../../../common/middlewares';
+import { UserService, userService } from '../../user/service/user.service';
+import { ICartController } from './cart.controller.interface';
+
+const cartUpdateSchema = Joi.object({
+	productId: Joi.string().uuid().required(),
+	count: Joi.number().integer().required(),
+});
+
+const cartCheckoutSchema = Joi.object({
+	payment: Joi.object({
+		type: Joi.string().required(),
+		address: Joi.string(),
+		creditCard: Joi.string(),
+	}),
+	delivery: Joi.object({
+		type: Joi.string().required(),
+		address: Joi.string(),
+	}),
+	comments: Joi.string().min(0).max(600),
+});
+
+export class CartControllerRest extends BaseController implements ICartController {
+	constructor(
+		private userService: UserService,
+		private cartService: CartService,
+	) {
+		super();
+		this.bindRoutes([
+			{
+				path: '/',
+				method: 'get',
+				func: this.getOne,
+				middlewares: [new AuthMiddleware(this.userService)],
+			},
+
+			{
+				path: '/',
+				method: 'put',
+				func: this.update,
+				middlewares: [
+					new ValidateMiddleware(cartUpdateSchema),
+					new AuthMiddleware(this.userService),
+				],
+			},
+
+			{
+				path: '/',
+				method: 'delete',
+				func: this.empty,
+				middlewares: [new AuthMiddleware(this.userService)],
+			},
+
+			{
+				path: '/checkout',
+				method: 'post',
+				func: this.checkout,
+				middlewares: [
+					new ValidateMiddleware(cartCheckoutSchema),
+					new AuthMiddleware(this.userService),
+				],
+			},
+		]);
+	}
+
+	public getOne = async (req: Request, res: Response): Promise<void> => {
+		try {
+			const userId = req.headers['x-user-id'] as string;
```

Comment:
For now it is ok to get this user in controller level like this. But in real scenario, as you already know, we should have a middleware that provides the current user information to the controllers.

Reply:
I will try to make it in the next modules